librz/analysis/var_global.c

```diff
@@ -263,13 +281,14 @@ RZ_API bool rz_analysis_var_global_rename(RzAnalysis *analysis, RZ_NONNULL const
 		RZ_LOG_ERROR("Global variable '%s' does not exist!\n", old_name);
 		return false;
 	}
-	RZ_FREE(glob->name);
-	glob->name = strdup(newname);
 
-	if (glob->flag_item) {
-		rz_flag_rename(analysis->flb.f, glob->flag_item, newname);
+	RzFlagItem *flag = rz_analysis_var_global_get_flag_item(glob);
+	if (flag) {
+		rz_flag_rename(analysis->flb.f, flag, newname);
```

Comment:
Just noticed that this should be checked and nothing done if this returns false.

Reply:
I don't think so. A global var without a flag can still be handled fine, so I think it should just do whatever it can here.

Reply:
I also agree, since it's somewhat like a hack solution to create flags for global variables. They are separate concepts and in the future we should be able to not create the flag for those, IMHO.